src/internal/_xgroupBy.js

```diff
@@ -0,0 +1,39 @@
+var _append = require('./_append');
+var _curry2 = require('./_curry2');
+
+module.exports = (function() {
+    function _xgroupBy(f, xf) {
+        return new XGroupBy(f, xf);
+    }
+
+    function XGroupBy(f, xf) {
+        this.xf = xf;
+        this.f = f;
+        this.inputs = {};
+    }
+
+    XGroupBy.prototype.init = function() {
+        return this.xf.init();
+    };
+
+    XGroupBy.prototype.result = function(result) {
+        var key;
+        for (key in this.inputs) {
+            result = this.xf.step(result, this.inputs[key]);
```

Comment:
should be *pretty* safe since `this.inputs` should really only be manipulated by the `XGroupBy` instance.